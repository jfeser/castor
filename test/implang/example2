select([p_counter, c_counter],
ahashidx(dedup(
      join(true, select([id as p_id], log), select([id as c_id], log))) as k,
  alist(select([p_counter, c_counter],
    join(p_counter < c_counter && c_counter < p_succ,
      filter(p_id = k.p_id,
        select([id as p_id, counter as p_counter, succ as p_succ], log)),
      filter(c_id = k.c_id,
        select([id as c_id, counter as c_counter], log)))) as lk,
    atuple([ascalar(lk.p_counter), ascalar(lk.c_counter)], cross)),
  (id_p, id_c)))
