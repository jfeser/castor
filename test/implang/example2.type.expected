       (FuncT
        (((HashIdxT
           ((TupleT
             (((IntT ((range (Interval 1 1)))) (IntT ((range (Interval 2 3)))))
              ((kind Cross))))
            (ListT
             ((TupleT
               (((IntT ((range (Interval 1 4)))) (IntT ((range (Interval 2 5)))))
                ((kind Cross))))
              ((count (Interval 1 2)))))
            ((key_count (Interval 2 2))))))
         (Width 2)))
